src/modules/headset/headset_openxr.c

```diff
@@ -1685,13 +1706,42 @@ static ModelData* openxr_newModelData(Device device, bool animated) {
   return model;
 }
 
-static bool openxr_animate(Device device, Model* model) {
-  XrHandTrackerEXT tracker = getHandTracker(device);
+static ModelData* openxr_newModelDataMSFT(XrControllerModelKeyMSFT modelKey, bool animated) {
+  uint32_t size;
+  if (XR_FAILED(xrLoadControllerModelMSFT(state.session, modelKey, 0, &size, NULL))) {
+    return NULL;
+  }
 
-  if (!tracker) {
-    return false;
+  unsigned char* modelData = malloc(size);
+  if (!modelData) return NULL;
+
+  if (XR_FAILED(xrLoadControllerModelMSFT(state.session, modelKey, size, &size, modelData))) {
+    free(modelData);
+    return NULL;
   }
 
+  Blob* blob = lovrBlobCreate(modelData, size, "Controller Model Data");
+  ModelData* model = lovrModelDataCreate(blob, NULL); // @TODO: null not allowed here
```

Comment:
Hm, maybe `NULL` could be allowed and the model loader would just throw if there were any references to external files.  This can be left for the future though.